ecs-cli/modules/cli/local/up_app.go

```diff
@@ -40,26 +43,50 @@ import (
 // If the container is not running, this command creates a new network for all local ECS tasks to join
 // and communicate with the Amazon ECS Local Endpoints container.
 func Up(c *cli.Context) {
-	// TODO When we don't provide any flags, Create() should just check if a "./docker-compose.local.yml" already exists.
-	// If so then Create() should do nothing else, otherwise it should error.
-	Create(c)
+	if err := options.ValidateCombinations(c); err != nil {
+		logrus.Fatalf(err.Error())
+	}
+	composePath, err := createComposeFile(c)
+	if err != nil {
+		logrus.Fatalf("Failed to get compose file path due to:\n%v", err)
+	}
+	runContainers(composePath)
+}
+
+// createComposeFile returns the path of the Compose file to start the containers from.
+// The Compose file ordering priority is defined as:
+// 1. Use the --task-def-compose flag if it exists
+// 2. Use the default docker-compose.local.yml file if no flags were provided and the file exists
+// 3. Otherwise, we need to create the Compose file.
+func createComposeFile(c *cli.Context) (string, error) {
+	if name := c.String(flags.TaskDefinitionCompose); name != "" {
+		return filepath.Abs(name)
+	}
+	if shouldUseDefaultComposeFile(c) {
```

Comment:
This is a little bit confusing to me: 

`local up` can either "create" and "run", or just "run". And the `-c` flag is supposed to mean "run", while without `-c` flag, `local up`  is supposed to "create" and then "run".

My suggest is to run the default `docker-compose.local.yml` file while user provide `-c` with no value. And the other part for `up` should be the same as `create` (e.g., `local up` create from `task-definition.json` and error out if no default file exists).

Reply:
Updated the PR based on today's discussion :)

Using `task-definition.json` over `docker-compose.local.yml` when no flags are provided.

Reply:
Great job! But will the default docker compose file be run (if exists) if `-c` is set with empty value? Or it should always error out?

Reply:
My understanding is  that it should error out because `-c`, `-r`, or `-f` are input values. If a flag is set without a value then it should be an error.